### PR TITLE
Create a feedback model with associations

### PR DESCRIPTION
**Pull request by GALTdea**
This pr is the first for the feedback feature

In this pr, we add a Feedback model with polymorphic associations.

### CODE REVIEW

File: db/migrate/20230522205955_create_feedbacks.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class CreateFeedbacks < ActiveRecord::Migration[7.0]
+  def change
+    create_table :feedbacks do |t|
+      t.references :author, null: false, foreign_key: {to_table: :users}
+      t.references :receiver, null: false, foreign_key: {to_table: :users}
+      t.references :referenceable, polymorphic: true, null: false
+      t.integer :overall_rating, default: 0, null: false
+      t.integer :status, default: 0, null: false
+      t.jsonb :data
```
Comment (toyhammered) [inline db/migrate/20230522205955_create_feedbacks.rb:9]: should default this to `{}` and null false.

File: app/models/feedback.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+#
+# Foreign Keys
+#
+#  fk_rails_...  (author_id => users.id)
+#  fk_rails_...  (receiver_id => users.id)
+#
+class Feedback < ApplicationRecord
+  belongs_to :author, class_name: "User"
+  belongs_to :receiver, class_name: "User"
+  belongs_to :referenceable, polymorphic: true
```
Comment (toyhammered) [inline app/models/feedback.rb:31]: should we also include the enum for status in this PR? (0 should be draft)

File: spec/factories/feedbacks.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+#
+# Foreign Keys
+#
+#  fk_rails_...  (author_id => users.id)
+#  fk_rails_...  (receiver_id => users.id)
+#
+FactoryBot.define do
+  factory :feedback do
+    author { nil }
+    receiver { nil }
```
Comment (toyhammered) [inline spec/factories/feedbacks.rb:31]: when you set these up, make sure to point it to a user (should be example in other factory bots).

**Review by toyhammered — COMMENTED** [review]
This looks amazing! Couple small things to fix but otherwise great job!

**Review by JoshDevHub — COMMENTED** [review]
Looks good to me. I'll tag @toyhammered just to get his approval for it.

**Pull request merged**